src/ThemeProvider.tsx

```diff
@@ -146,17 +146,18 @@ function chooseColorScheme(colorMode: ColorMode, dayScheme: string, nightScheme:
   }
 }
 
-function applyColorScheme(theme: Theme, colorScheme: string) {
+function applyColorScheme(theme: ThemePrevalType, colorScheme: string): Theme {
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
   if (!theme.colorSchemes) {
-    return theme
+    return theme as Theme
   }
 
   if (!theme.colorSchemes[colorScheme]) {
     // eslint-disable-next-line no-console
     console.error(`\`${colorScheme}\` scheme not defined in \`theme.colorSchemes\``)
 
     // Apply the first defined color scheme
-    const defaultColorScheme = Object.keys(theme.colorSchemes)[0]
+    const defaultColorScheme = Object.keys(theme.colorSchemes)[0] as keyof ThemePrevalType['colorSchemes']
```

Comment:
just to make the type access at theme.colorSchemes[defaultColorScheme] happy - but should improve this access